sunpy/map/mapbase.py

```diff
@@ -265,7 +265,15 @@ def wcs(self):
         """
         The `~astropy.wcs.WCS` property of the map.
         """
-        w2 = astropy.wcs.WCS(naxis=2)
+        # Construct the WCS based on the FITS header, but don't "do_set" which
+        # analyses the FITS header for correctness.
+        w2 = astropy.wcs.WCS(header=self.fits_header, _do_set=False)
```

Comment:
This seems to be changing the way the `wcs` is computed quite substantially, ie. now it does it from the header instead of from just the map properties (below)